### PR TITLE
feat: add endpoint to retrieve onboarding repos based on the PAT

### PR DESCRIPTION
**Pull request by brandonroberts**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

Pulls the available repos for a given Personal Access Token as an array of strings

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -71,4 +72,32 @@ export class AuthController {
...
+    @Body() body: string,
+  ): string[] {
+    const { token } = JSON.parse(body || "{}") as { token: string | undefined };
+
+    if (!token) {
+      throw (new NotFoundException);
+    }
+
+    try {
+      const repoList = execSync(`npx repo-report ls --token="${token}"`).toString("utf-8");
```
Comment (0-vortex) [inline src/auth/auth.controller.ts:95]: We can't spawn on droplets but you can use the `getRepositories` function with normal await - it's just gonna do a graphql request :D

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -71,4 +72,32 @@ export class AuthController {
...
+  @ApiOperation({
+    operationId: "postOnboardingRepos",
+    summary: "Retrieves onboarding repos using a PAT",
+  })
+  @ApiOkResponse()
+  @HttpCode(HttpStatus.OK)
+  postOnboardingRepos (
+    @Body() body: string,
+  ): string[] {
+    const { token } = JSON.parse(body || "{}") as { token: string | undefined };
```
Comment (0-vortex) [inline src/auth/auth.controller.ts:88]: I think you need the `@User() user: SupabaseAuthUser` decorator here

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -71,4 +72,32 @@ export class AuthController {
...
+
+    if (!token) {
```
Comment (0-vortex) [inline src/auth/auth.controller.ts:90]: this block is redundant

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Some minor changes here and there

File: src/auth/auth.controller.ts (hunk 2 above)
Comment (brandonroberts) [reply src/auth/auth.controller.ts:88]: Is this so it will force an authenticated session to be provided?

Comment (0-vortex) [reply src/auth/auth.controller.ts:88]: Yes, we can't let users ddos this API out by providing a custom broken tokens, but we can store a token encrypted for the logged in user and then re-use it

**Pull request closed without merging**